### PR TITLE
Update dependency eslint-config-amo to v5.2.0

### PR DESCRIPTION
**Pull request by willdurand**
Closes #4041

---

I disabled a new rule in two cases because rewritting the regexp without
`RegExp` when it's dealing with a path isn't very readable.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { PACKAGE_TYPES, LOCAL_PROTOCOLS } from 'const';
 
 /* global nodeRequire, localesRoot */
 
-const SOURCE_MAP_RE = new RegExp(/\/\/[#@]\s(source(?:Mapping)?URL)=\s*(\S+)/);
+const SOURCE_MAP_RE = /\/\/[#@]\s(source(?:Mapping)?URL)=\s*(\S+)/;
```
Comment (willdurand) [inline src/utils.js:14]: My understanding is that `new RegExp` was useless here, its only argument is already a regexp.

**Comment by willdurand** [comment]
Let's forget about the codecov check, I didn't change any code coverage.

File: src/schema/firefox-schemas-import.js [hunk 2]
```diff
@@ -32,7 +32,9 @@ const UNRECOGNIZED_PROPERTY_REFS = [
 ];
 
 const schemaRegexes = [
+  // eslint-disable-next-line prefer-regex-literals
   new RegExp('browser/components/extensions/schemas/.*\\.json'),
+  // eslint-disable-next-line prefer-regex-literals
   new RegExp('toolkit/components/extensions/schemas/.*\\.json'),
```
Comment (rpl) [inline src/schema/firefox-schemas-import.js:38]: Any reason for not changing this too? I guess too many '\' to escape? (anyway fine by me to disable it here)

**Review by rpl — APPROVED** [review]

File: src/schema/firefox-schemas-import.js (hunk 2 above)
Comment (willdurand) [reply src/schema/firefox-schemas-import.js:38]: Yes, escaping backslashes in a path doesn't make the regexp very readable.

**Pull request merged**